Permission errors for precompilation on Windows

Several users report file permission errors when precompiling on Julia 1.5.3 which did not happen on 1.5.2, see https://discourse.julialang.org/t/precompilation-error-with-julia-1-5-3-windows-10-64-bit/50028. Since there was a libuv upgrade (rebuild?) between those to releases I guess that might be related.

encountered this permission error during the compiling process in cygwin-64. The julia.exe excuatable has been generated successfully and then failed in the julia precompilation.

```
make[1]: Entering directory '/cygdrive/c/Users/pem/Downloads/julia-1.5.3/julia-win64'
Warning: git information unavailable; versioning information limited
    JULIA julia-win64/usr/lib/julia/sys-o.a
ERROR: LoadError: IOError: mkdir: permission denied (EACCES)
Stacktrace:
 [1] uv_error at .\libuv.jl:97 [inlined]
 [2] mkdir(::String; mode::UInt16) at .\file.jl:177
 [3] mkpath(::String; mode::UInt16) at .\file.jl:227
 [4] mkpath(::String; mode::UInt16) at .\file.jl:225
 [5] mkpath at .\file.jl:222 [inlined]
 [6] (::Main.anonymous.var"#1#5")(::String) at C:\Users\pem\Downloads\julia-1.5.3\contrib\generate_precompile.jl:76
 [7] mktempdir(::Main.anonymous.var"#1#5", ::String; prefix::String) at .\file.jl:709
 [8] mktempdir at .\file.jl:707 [inlined] (repeats 2 times)
 [9] generate_precompile_statements() at C:\Users\pem\Downloads\julia-1.5.3\contrib\generate_precompile.jl:72
 [10] top-level scope at C:\Users\pem\Downloads\julia-1.5.3\contrib\generate_precompile.jl:202
in expression starting at C:\Users\pem\Downloads\julia-1.5.3\contrib\generate_precompile.jl:3
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [/cygdrive/c/Users/pem/Downloads/julia-1.5.3/sysimage.mk:86: /cygdrive/c/Users/pem/Downloads/julia-1.5.3/julia-win64/usr/lib/julia/sys-o.a] Error 1
make[1]: Leaving directory '/cygdrive/c/Users/pem/Downloads/julia-1.5.3/julia-win64'
make: *** [/cygdrive/c/Users/pem/Downloads/julia-1.5.3/Makefile:87: julia-sysimg-release] Error 2
make: Leaving directory '/cygdrive/c/Users/pem/Downloads/julia-1.5.3/julia-win64'

```

--- a/base/file.jl
+++ b/base/file.jl
@@ -179,6 +179,8 @@ function mkdir(path::AbstractString; mode::Integer = 0o777)
             uv_error("mkdir($(repr(path)); mode=0o$(string(mode,base=8)))", ret)
         end
         ccall(:uv_fs_req_cleanup, Cvoid, (Ptr{Cvoid},), req)
+        # mode is not implemented in mkdir in libuv yet, so do it here
+        Sys.iswindows() && chmod(path, mode)
         return path
     finally
         Libc.free(req)

